crates/uv-resolver/src/redirect.rs

```diff
@@ -36,9 +39,50 @@ pub(crate) fn apply_redirect(url: &VerbatimUrl, redirect: Url) -> VerbatimUrl {
     redirect
 }
 
+pub(crate) fn url_to_precise(url: VerbatimParsedUrl) -> VerbatimParsedUrl {
+    let ParsedUrl::Git(ParsedGitUrl {
+        url: git_url,
+        subdirectory,
+    }) = url.parsed_url.clone()
+    else {
+        return url;
+    };
+
+    // TODO(konsti): Remove once we carry more context on the `Dist`s.
+    let lowered_git_ref = git_url
+        .reference()
+        .as_str()
+        .map_or(GitReference::DefaultBranch, |rev| {
+            GitReference::from_rev(rev)
+        });
+    let git_url = GitUrl::new(git_url.repository().clone(), lowered_git_ref);
+
+    let Some(new_git_url) = git_url_to_precise(git_url.clone()) else {
+        if matches!(git_url.reference(), GitReference::FullCommit(_)) {
+            return url;
+        }
+        panic!("Unseen git url: {}, {:?}", url.verbatim, git_url);
```

Comment:
Should we just return the original URL here instead?

Reply:
Made this a debug assert, this would be a bug in uv but not one we have to concern the user with (we only fail the already-install checked if tha happens).